clang/test/dpct/cu_mem_alloc_typedef.cu

```diff
@@ -0,0 +1,12 @@
+// UNSUPPORTED: system-linux
+// RUN: dpct --format-range=none -out-root %T/cu_mem_alloc_typedef %s --cuda-include-path="%cuda-path/include" -- -std=c++14 -x cuda --cuda-host-only
+// RUN: FileCheck %s --match-full-lines --input-file %T/cu_mem_alloc_typedef/cu_mem_alloc_typedef.dp.cpp
+#include <cstdint>
+#include <cuda.h>
+
+typedef uint64_t CUdeviceptr;
```

Comment:
CUdeviceptr is user user-defined type, in L9 and L11,  DPCT should keep it without migration.  (As it is user-defined type).

Reply:
can we find all the other typedefs?  (like the typedef in cuda header file). If so, we can migrate it..

Reply:
If there is a typedef and it's present in CUDA headers, and if nvcc can compile the code -- it mean the underlying types is same everywhere. I think it will be redundant to look for the type in CUDA header as we rely on the fact that for a successful migration the CUDA source should be compilable. Thoughts?